src/compiler/resolutionCache.ts

```diff
@@ -50,19 +54,37 @@ namespace ts {
         const resolvedTypeReferenceDirectives = createMap<Map<ResolvedTypeReferenceDirectiveWithFailedLookupLocations>>();
 
         const directoryWatchesOfFailedLookups = createMap<DirectoryWatchesOfFailedLookup>();
+        const failedLookupLocationToDirPath = createMap<Path>();
+        let rootDir: string;

```

Comment:
Why do we need both? When would we ever need the version that *hasn't* been normalized to a path?

Reply:
Also, it looks like bad things will happen if the user forgets to call `setRootDirectory`? Could we design the API so they have to provide that information so we can call `setRootDirectory` when initializing?